packages/ui/src/server/controllers/vcs.ts

```diff
@@ -0,0 +1,261 @@
+import {AppRouteHandler, Response} from '@gravity-ui/expresskit';
+import {AxiosError} from 'axios';
+import {VcsConfig, VcsRepository} from '../../shared/vcs';
+import {sendError} from '../utils';
+import {VcsFactory} from '../components/vcs';
+
+const COOKIE_OPTIONS = {
+    httpOnly: true,
+    secure: true,
+    sameSite: 'strict' as const,
+    maxAge: 1000 * 60 * 60 * 24 * 365,
+};
+
+const ERROR_MESSAGE = {
+    VCS_CONFIG_NOT_FOUND: 'vcsSettings is not found in UISettings',
+    VCS_ID_REQUIRED: 'vcsId is required',
+    VCS_NOT_SUPPORTED: 'The current vcsId is missing from the vcsSettings setting in UISettings',
+    TOKEN_REQUIRED: 'Token is required',
+    PATH_REQUIRED: 'Path is required',
+    BRANCH_REQUIRED: 'Branch is required',
+    REPOSITORY_REQUIRED: 'Repository is required',
+    UNAUTHORIZED:
+        'The service cannot authorize you. Check the token. You can add new token in the section Settings -> VCS',
+};
+
+const cookieNameByVcsId = (vcsId: string) => `vcs_${vcsId}`;
+const sendApiError = (res: Response, error: AxiosError) => {
+    let status = error.response?.status;
+    let message = error.response?.statusText;
+
+    if (status === 401) {
+        status = 500;
+        message = ERROR_MESSAGE.UNAUTHORIZED;
+    }
+
+    sendError(res, message || error.message, status || 500);
+};
+
+export const getVcsConfig: AppRouteHandler = (req, res) => {
+    const cookies = req.cookies;
+    try {
+        const config = VcsFactory.getVcsSettings(req);
+        if (!config.length) {
+            sendError(res, ERROR_MESSAGE.VCS_CONFIG_NOT_FOUND, 404);
+            return;
+        }
+
+        const response = config.map<VcsConfig>((item) => ({
+            ...item,
+            token: cookieNameByVcsId(item.id) in cookies,
```

Comment:
It is better to avoid sending of tokens as is. But we can send only first and last letters like:
`some_long_user_token` -> `som****ken`

Reply:
const cookieNameByVcsId = (vcsId: string) => `vcs_${vcsId}`;

the token itself is not passed there. It looks like I should replace the variable name as it might be misleading